File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.cpp
#include <bits/stdc++.h>

using namespace std;

int gcd(int a, int b){
    if(b == 0) return a;
    else return gcd(b, a % b);
}

int solution(vector<int> arr) {
    int answer = 0;
    
    int tmp = 1;
    for(int i = 0; i < arr.size(); i++){
        tmp = tmp * arr[i] / gcd(tmp, arr[i]);
    }
    return tmp;
}

File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.py
from functools import reduce

def solution(arr):    
    return reduce(lcm , arr, 1)

def euclidean(a, b):
    while b != 0:
        r = a % b
        a = b
        b = r
    return a

def lcm(x, y):
    return x * y / euclidean(x, y)